pkg/doc/verifiable/credential.go

```diff
@@ -316,6 +378,126 @@ func NewCredential(dataJSON []byte, opts ...CredentialOpt) (*Credential, error)
 	return cred, nil
 }
 
+// credentialClaims is JWT Claims extension by Credential.
+type credentialClaims struct {
+	*jwt.Claims
+
+	Credential *rawCredential `json:"vc,omitempty"`
+}
+
+// rawCredentialClaims is used to get raw content of "vc" claim of JWT.
+type rawCredentialClaims struct {
+	*jwt.Claims
+
+	Raw map[string]interface{} `json:"vc,omitempty"`
+}
+
+func decodeJws(rawJwt []byte, fetcher PublicKeyFetcher) ([]byte, *rawCredential, error) {
+	parsedJwt, err := jwt.ParseSigned(string(rawJwt))
+	if err != nil {
+		return nil, nil, fmt.Errorf("VC is not JWS: %w", err)
+	}
+
+	var credClaims credentialClaims
+	err = parsedJwt.UnsafeClaimsWithoutVerification(&credClaims)
```

Comment:
Now that we agreed to always verify the signature, I don't think we should be using any of these "UnsafeXXX" functions (unless I'm mistaken)

Reply:
That's because of an indirect algorithm: we need to read **Issuer** JWT claim first, find the proper public key (by **Issuer (DID)** and **Key ID**) and only then verify the signature.

Reply:
Ok, got it.

This is kinda funky: we shouldn't be touching these claims at all without verification. But, on the other hand, including the sender's identifier in the `kid` will reveal it to eavesdroppers... not sure what can be done about it.